src/Sulu/Bundle/MediaBundle/Resources/js/containers/MediaSelection/MediaSelectionOverlay.js

```diff
@@ -0,0 +1,241 @@
+            open,
+            locale,
+        } = this.props;
+
+        if (open) {
+            this.overlayDisposer = autorun(() => this.createStores(locale));
+        }
+    }
+
+    componentWillReceiveProps(nextProps: Props) {
+        const {
+            open,
+            locale,
+        } = this.props;
+
+        if (!open && nextProps.open) {
+            this.overlayDisposer = autorun(() => this.createStores(locale));
+        }
+    }
+
+    @action destroy() {
+        this.destroyStores();
+        this.overlayDisposer();
+
+        this.selectedMedia = [];
+        this.collectionId = undefined;
+    }
+
+    @action setMediaPage(page: number) {
+        this.mediaPage.set(page);
+    }
+
+    @action setCollectionPage(page: number) {
+        this.collectionPage.set(page);
+    }
+
+    @action setCollectionId(id: ?string | number) {
+        this.collectionId = id;
+    }
+
+    createStores = (locale: observable) => {
+        this.setMediaPage(1);
+        this.setCollectionPage(1);
+
+        this.createCollectionStore(this.collectionId, locale);
+        this.createMediaDatagridStore(this.collectionId, this.mediaPage, locale);
+        this.createCollectionDatagridStore(this.collectionId, this.collectionPage, locale);
+    };
+
+    destroyStores() {
+        if (this.collectionStore) {
+            this.collectionStore.destroy();
+        }
+
+        if (this.mediaDatagridStore) {
+            this.mediaDatagridStore.destroy();
+        }
+
+        if (this.collectionDatagridStore) {
+            this.collectionDatagridStore.destroy();
+        }
+    }
+
+    @action createCollectionDatagridStore(collectionId: ?string | number, page: observable, locale: string) {
+        if (this.collectionDatagridStore) {
+            this.collectionDatagridStore.destroy();
+        }
+
+        this.collectionDatagridStore = new DatagridStore(
+            COLLECTIONS_RESOURCE_KEY,
+            {
+                page,
+                locale,
+            },
+            (collectionId) ? {parent: collectionId} : undefined
+        );
+    }
+
+    createCollectionStore = (collectionId: ?string | number, locale: string) => {
+        if (this.collectionStore) {
+            this.collectionStore.destroy();
+        }
+
+        this.collectionStore = new CollectionStore(collectionId, locale);
+    };
+
+    @action createMediaDatagridStore(collectionId: ?string | number, page: observable, locale: string) {
+        const {excludedIds} = this.props;
+        const options = {};
+
+        options.fields = [
+            'id',
+            'type',
+            'name',
+            'size',
+            'title',
+            'mimeType',
+            'subVersion',
+            'thumbnails',
+        ].join(',');
+
+        page.set(1);
+
+        if (collectionId) {
+            options.collection = collectionId;
+        }
+
+        if (excludedIds.length) {
+            options['excluded-ids'] = excludedIds.join(',');
```

Comment:
We've decided to go for `excluded` here, because we have that already in other requests, and we would like to stay consistent.